### ISSUE TITLE
[Character - Concept in Progress] Hannah

### ISSUE THREAD

**Issue by Floofball** [ProbablyTez/Agony#398]
Hannah is unlocked through beating any run where no Devil Deals are taken and no Devil Beggars are donated to. The hint on the playing screen will be fairly simple: "Stay faithful".

So, what exactly sets Hannah apart from your average Isaac character? Well, there are certainly a few things, and the big one is to make Angel Rooms more useful and fun for the player ...

Starting Stats
_______________________________________________________________________________________________________________
- HP - 3 Red Containers, 1 Spirit
- Tears - Above Avg.
- Dmg - 1.5
- Speed - Base
- Range - Cain
- Shot Speed - Base
- Luck - -1

Starting Items
_______________________________________________________________________________________________________________

Active Item - n/a

Passive Item - Saint's Hood

Mechanics
_______________________________________________________________________________________________________________

Hannah has poor offensive power, but starts out with an (initially) exclusive item: Saint's Hood, which grants a passive x2 dmg multiplier, raising her initially poor dmg stat to 3 (near to Isaac's base), as well as providing her with the benefit of Devil/Angel rooms even if she takes red heart damage. However, if Hannah. takes any Devil Deals, that item, and thus those benefits, are taken away. This means that paying for Brimstone on the second floor sets you back to 1.5 dmg and thus puts you into a rather precarious position. Now, this may seem like its taking away the variety of Devil Deals away from the game if you want to succeed, and I would agree were it not for: a) Most Isaac players default to Devil Deals, and often see it as the right way to play, seeing as Angel Rooms are generally poorer and more difficult to get due to how good Devil Deals are (which Saint's Hood helps remedy), and b) most players who choose Devil Deals over Angel Rooms have to force themselves to do Angel runs (which Saint's Hood offers an incentive for), thus making Hannah not only a welcome addition for those who wish for more interesting and diverse playthroughs of Isaac, but also from the perspective of strategists and minmaxers, who may enjoy the added layer of consequence and strategy.

As for unlocks, I'm working on ideas right now. If I or any of you come up with any, I'll make sure to edit the post and include them!

Edit:

Unlocks:
_______________________________________________________________________________________________________________

- [Saint's Hood] "Blessed is the mother who prays" - Beat Satan with Hannah

- [Gospel Child] (https://github.com/AgonyTeam/Agony/issues/482) "Sing for the choir" - Beat The Lamb with Hannah

- [Isaac] "Prove your devotion" - Beat Isaac with Hannah; Isaac is a familiar which acts similar to the Mulligan, but enemies do not attract it, as well as grants a +2 Tear Delay. If you can goad enemies into contact with Isaac, he will take damage and, eventually, die. Upon dying, Hannah will be granted 2 random Angel Room passives as well as a -5 Tear Delay.

- [Holy Oath] "Journey to the Promised Land" - Beat ??? with Hannah; Holy Oath, upon pickup, grants an all stats down, however, if the player does not take Devil Deals for 4 consecutive floors, Hannah is granted 3 random Angel Room items (2 if any Devil Deals have been taken before picking up the Oath) and 4 Spirit Hearts (2 if Devil Deals were taken)

- ??? "???" - Beat Boss Rush with Hannah; ???

- ??? "???" - Beat Ultra Greed with Hannah; ???

- ??? "???" - Beat Hush with Hannah; ???

- ??? "???" - Beat Mom with Hannah; ???

- ??? "???" - Beat Mom's Heart/It Lives with Hannah; ???

- ??? "???" - Beat Ultra Greedier with Hannah; ???

- ??? "???" - Beat Delirium with Hannah; ???

- ??? "???" - Beat Mega Satan with Hannah; ???

- ??? "???" - Beat Ezekiel with Hannah; ???

**Comment by Anonymous-Koala** [comment]
Interesting, although this creates a new problem. Angel Deals need to be viable enough(at least with Abraham) for Faith's Reward to be incentivized, otherwise players just choose to play a handicapped character who needs Angel Deals to actually have decent damage.

As for unlocks, Binding of Isaac(the biblical tale) needs to be referenced. Either an Isaac familiar who you're incentivized to allow the death of in some way for a benefit(Probably something angel themed), or a Lamb fitting to how in the end they ended up killing a Lamb instead of Isaac.

I'm going to try going through my biblical knowledge to think of some more unlocks.

**Comment by Floofball** [comment]
@Anonymous-Koala First off, thanks for the input! Your unlock ideas are really great! I'll make sure to tweak them and add a bit more detail before adding them to the post!

Secondly, to address the problem you brought up; the only simple way I can see it solved without making Abraham overpowered is to make Angel Rooms better in general. Rework the rooms and/or even remove some items from the pool (Breath of Life, for example, is totally useless. Maybe have a specific Eternal Chest item pool?). Additionally, seeing as Abraham already starts with the Bible, that FORTUNATELY is not a problem, but Angel Rooms item pools/rooms just need to be reworked in general.

As an idea, maybe every Angel Room contains an item? Rather than having 1 or 3 or whatever number of Eternal Chests/Spirit Hearts, just have those as bonuses. That way, Angel Rooms can be more reliable and thus more viable not only for other characters, but especially for Abraham.

Additionally, other characters have plenty of handicaps. Judas starts with only 1 HP, ??? has only Spirit Hearts, etc., and those things generally, unless you are an extreme minmaxer, rely more so on your luck than your skill, if only by a little, so I see no problem with having Abraham's staple be Angel Rooms. It gives the game a bit more dimension. You may get an AMAZING Devil Deal on the second floor with Goat Head, Maw of the Void, and Brimstone, and thus have to weight out your options of whether to take the items then or forsake them for the possibility of Sacred Heart, Godhead, etc. (which you often already have to do if the game hates you and you have low/no HP) but in this case makes that more of a choice than "Well I only got 2 spirit hearts instead of 12, guess it's Angel Rooms for me.", etc. etc.

Those are just my thoughts though, to be honest. I could be completely wrong and it turns out Abraham is one of the shittiest characters in the game, but game design is just kind of life that sometimes, I guess.

**Comment by Anonymous-Koala** [comment]
Eternal Chests need to be unable to result with nothing on the first opening(or the first few). Breath of Life and Dead Sea Scrolls are absolutely shit and with how rare Angel Deals are/how Devil Deals are always better, removing shit items isn't gonna create an unbalanced pool. That, and not paying out with items enough, are the Angel Deals' shortfall. The Devil Deal is a constant that you can always trust to give you something powerful, but Angel Deals(Outside of Antibirth) are anything but reliable.

I'm not sure if Abraham should start with the bible. Although this does work gameplay wise, meaning you will not see it again unless you pick up the Rosary. 

Also a way to make Angel Deals more viable, without making them as reliable as Devil Deals(though they still need to be more reliable, right now it's way too random), is to give Abraham, or have this from one of his items, There's Options(or stronger, either by letting you take both, or giving you more options to see) but for Angel Deals.

**Comment by Floofball** [comment]
@Anonymous-Koala Hmm, good idea! I'll add that effect to Faith's Reward in a minute.

Alsooo, as for the Bible, it fits gameplay and lorewise (to the Character, at least), seeing as Abraham was incredibly faithful to God. Additionally, it allows the player to have easier Mom/Mom's Heart fights which could be seen as a sort of boost due to the difficulty of unlocking the character.

Honestly, the problem I see with Abraham right now is that he can go either one of two ways, a) be super OP, or b) be super UP.

**Comment by Anonymous-Koala** [comment]
Actually, that would be a better effect for Isaac. God's Favor after appeasing him by killing the Isaac familiar you unlock(Maybe Abraham could start with it once unlockd?), is that effect for the floor. To balance that out killing Isaac has to have some problem, either be difficult to pull off or with some negative effect.

I think that isn't really a problem. It worked for Bethanny and it's interesting because it perfectly captures what Angel Deals are like. They can be powerful, but they can be incredibly bad or just not deliver when you need them too. And that's also why she works, because Angel Deals in Antibirth are an example of what happens when the reward offsets the risk. In AB+ there's a lot of risk but not a lot of rewards that are worth it. Once the reward is upped and the deals become more reliable to a degree, an angel deal themed character could definetly work.

**Comment by Floofball** [comment]
Good point, although I don't really get why it'd be a better effect for Isaac. Isaac getting a favor from God for killing a mini version himself just seems a tad ... strange. Also, as for killing him, he could function somewhat like Mulligan where he attempts to run away from monsters, only they still focus on you rather than him, making it so you'd have to bait them with yourself to get him hit.

**Comment by Anonymous-Koala** [comment]
I think it fits with the theme. Isaac imagines this story of how God wants him to die, would fit to have an item that actually acts like that.

I like that idea though, makes killing him bothersome enough that it wouldn't be overpowered.

**Comment by Floofball** [comment]
Hmm, yeah I guess that works well enough. And all the characters (with the exception of Lilith, ???, Keeper, and The Lost) are just Isaac dressing up anyway, so

**Comment by Anonymous-Koala** [comment]
You wrote Lilith instead of Azazel, just so you know.

As for more unlocks, he could have a reference to the "Promised Land", which the quest for has started with Abraham as he's the first Jew. Some stat boost(either boost to random stats, all stats up, or an all stats up overall which means every time the boost is applied its part of the overall package) is applied once you start the next floor after picking this item up.

**Comment by Floofball** [comment]
Agh, duh. Sorry, shoulda said both Lilith AND Azazel. At least, I think Lilith isn't Isaac? I could be wrong, though.

**Comment by Anonymous-Koala** [comment]
Nah, Lilith is Isaac. Azazel is too, however. Only Blue Baby, Lost and maybe Keeper? are Isaac, as they're all different forms of his corpse. The rest of the chars are imaginated by him. 

How's the idea I suggested for another unlock?

**Comment by Floofball** [comment]
Sorry for responding so late. Had to go to sleep.

Anyway, the item concept is neat, but it feels a bit lacking. To make it a more interesting item, maybe have it give a stats DOWN but if you make it a certain amount of floors without taking Devil Deals you get a larger stat up than down, as well as 2-3 Angel Room items.

**Comment by Anonymous-Koala** [comment]
That would be interesting. Also means that its worth is based on how late in do you pick it up.

**Comment by ProbablyTez** [comment]
This character is kind of interesting. Eventhough it's not a completely new style of play, it's different enogh that it might just work.

Regarding the balancing of the Angel pool, Angel Items in Agony are pretty powerful and raise the level quite a bit. I think having shitty items in the pool isnt a bad thing though, since it creates an interesting problem mechanic wise (do I pass up on these mediocre devil items to risk a big reward on an angel deal ?) and mechanic wise (God tests faith)
